core/reserve_core.go

```
@@ -232,26 +236,43 @@ func (self ReserveCore) Withdraw(
 	return uid, err
 }
 
-func (self ReserveCore) pendingSetrateInfo(minedNonce uint64) (*big.Int, *big.Int, error) {
-	act, err := self.activityStorage.PendingSetrate(minedNonce)
+func calculateNewGasPrice(old *big.Int, count uint64) *big.Int {
+	// in this case after 5 tries the tx is still not mined.
+	// at this point, 50.1 gwei is not enough but it doesn't matter
+	// if the tx is mined or not because users' tx is not mined neither
+	// so we can just increase the gas price a tiny amount (1 gwei) to make
+	// the node accept tx with up to date price
+	if count > 4 {
+		return old.Add(old, common.GweiToWei(1))
+	} else {
+		// new = old + (50.1 - old) / (5 - count)
+		return old.Add(
+			old,
+			big.NewInt(0).Div(big.NewInt(0).Sub(common.GweiToWei(HIGH_BOUND_GAS_PRICE), old), big.NewInt(int64(5-count))),
+		)
+	}
+}
+
+func (self ReserveCore) pendingSetrateInfo(minedNonce uint64) (*big.Int, *big.Int, uint64, error) {
+	act, count, err := self.activityStorage.PendingSetrate(minedNonce)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, 0, err
 	}
-	if act != nil {
-		nonce, _ := strconv.ParseUint(act.Result["nonce"].(string), 10, 64)
-		gasPrice, _ := strconv.ParseUint(act.Result["gasPrice"].(string), 10, 64)
-		return big.NewInt(int64(nonce)), big.NewInt(int64(gasPrice)), nil
-	} else {
-		return nil, nil, nil
+	if act == nil {
+		return nil, nil, 0, nil
 	}
+	nonce, _ := strconv.ParseUint(act.Result["nonce"].(string), 10, 64)
+	gasPrice, _ := strconv.ParseUint(act.Result["gasPrice"].(string), 10, 64)
```

should we handle the errors here?

---

We can ignore the error here because the nonce and gas price is enforced in other scope already.